components.yaml

```
@@ -5,54 +5,54 @@ GEOSadas:
 env:
   local: ./@env
   remote: ../ESMA_env.git
-  tag: v1.4.3
+  tag: v3.2.1
   develop: main
 
 cmake:
   local: ./@cmake
   remote: ../ESMA_cmake.git
-  tag: v1.0.11
+  tag: v3.4.2
   develop: develop
 
 ecbuild:
   local: ./@cmake/@ecbuild
   remote: ../ecbuild.git
-  tag: geos/v1.0.0
+  tag: geos/v1.0.6
 
 NCEP_Shared:
   local: ./src/Shared/@NCEP_Shared
   remote: ../NCEP_Shared.git
-  tag: v1.1.0
+  tag: rt-GEOSadas-5_29_0_prep3
   develop: main
 
 GMAO_Shared:
   local: ./src/Shared/@GMAO_Shared
   remote: ../GMAO_Shared.git
-  tag: cvs/GEOSadas-5_27_1_p3
+  tag: rt-GEOSadas-5_29_0_prep3
   develop: dev/adas
 
 MAPL:
   local: ./src/Shared/@MAPL
   remote: ../MAPL.git
-  tag: v1.1.14
+  tag: v2.6.7
   develop: develop
 
 FMS:
   local: ./src/Shared/@FMS
   remote: ../FMS.git
-  tag: geos/orphan/v1.0.3
+  tag: geos/2019.01.02+noaff.7 
   develop: geos/release/2019.01
 
 GEOSana_GridComp:
   local: ./src/Components/@GEOSana_GridComp
   remote: ../GEOSana_GridComp.git
-  tag: v1.3.0
+  tag: rt-GEOSadas-5_29_0_prep3
   develop: develop
 
 GEOSgcm_GridComp:
   local: ./src/Components/@GEOSgcm_GridComp
   remote: ../GEOSgcm_GridComp.git
-  tag: cvs/GEOSadas-5_27_1_p3
+  tag: rt-GEOSadas-5_29_0_prep3
```

```suggestion
  tag: v1.21.2
```
This should be v1.21.1 as that had all the needed commits for GEOSadas. It does have additional fixes to make sure Surface parameters are echoed out during the model run (which was inspired by a bug found by @rtodling during MAPL 2 ADAS testing)

https://github.com/GEOS-ESM/GEOSgcm_GridComp/compare/rt-GEOSadas-5_29_0_prep3..v1.12.2

---

The changes in the NCEP_Shared and those in GEOSgcm_App seem reasonable.

As for GEOSgcm_GridComp I will take your work for the changes being zero-diff; the default of ATM_CO2 changing from 0 to 2 seems meaningless ... the only value that changes things for atm_co2 is 4; any other number will get the code doing exactly the same thing - it seems ... unless something really hidden is happening in this part of the code wrt this variable.

---

`ATM_CO2` only matters for CatchCN, so it's "harmless" unless you are running that:

https://github.com/GEOS-ESM/GEOSgcm_GridComp/blob/b3176bc8b2c2422d90de4432500f1b7dd103c8c0/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Shared/GEOS_SurfaceGridComp.rc#L187